dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/weld/WebSphereCDIDeploymentImpl.java

```diff
@@ -725,11 +727,21 @@ public Collection<URL> getUnversionedBeansXmlURLs() {
             }
 
             URL parsedURL = beansXml.getUrl();
-            //if the URL is null then that means it was an empty beans.xml file
+            //if the URL is null then this may mean it was an empty beans.xml file
             //note that this may be an undocumented "feature" of the Weld SPI
             if (parsedURL == null) {
                 continue;
             }
+            //check if the file is really empty ... just whitespace like a single space would cause Weld to complain anyway
+            try (InputStream is = parsedURL.openStream()) {
+                if (is.available() == 0) {
+                    //file is empty
+                    continue;
+                }
+            } catch (IOException e) {
+                //could not read the file, assume it is empty
+                continue;
+            }
```

Comment:
Need FFDCIgnore here?

Note that you need to ignore `Throwable.class` when using try-with-resources.

Reply:
I can't think of a legitimate reason why we would get an IOException at this point ... so I think we want the FFDC?

Reply:
You still want to ignore `Throwable.class` because you're using try-with-resources.

The compiler generates synthetic catch blocks and the instrumentation isn't smart enough to tell the difference yet.